src/KeyVault/KeyVault/Commands/RemoveAzureKeyVault.cs

```diff
@@ -180,9 +184,16 @@ public override void ExecuteCmdlet()
                         VaultName,
                         () =>
                         {
-                            KeyVaultManagementClient.PurgeVault(
-                                vaultName: VaultName,
-                                location: Location);
+                            if (Hsm.IsPresent)
+                            {
+                                //PurgeManagedHsm();
```

Comment:
I'm thinking about throwing an exception saying "this is not yet supported" 🤔 Otherwise the cmdlet will just fail silently